apps/api/src/server/applications/subscriptions/subscriptions.controller.js

```diff
@@ -0,0 +1,144 @@
+import * as subscriptionRepository from '../../repositories/subscription.respository.js';
+import { eventClient } from '../../infrastructure/event-client.js';
+import { EventType } from '@pins/event-client';
+import { NSIP_SUBSCRIPTION } from '../../infrastructure/topics.js';
+import { buildSubscriptionPayload } from './subscriptions.js';
+import logger from '../../utils/logger.js';
+import { Prisma } from '@prisma/client';
+
+/**
+ * @type {import('express').RequestHandler}
+ * @throws {Error}
+ * @returns {Promise<void>}
+ */
+export async function getSubscription(request, response) {
+	const { query } = request;
+
+	const caseReference = String(query.caseReference);
+	const emailAddress = String(query.emailAddress);
+
+	try {
+		const res = await subscriptionRepository.findUnique(caseReference, emailAddress);
+
+		if (res === null) {
+			response.status(404).send({ errors: { notFound: 'subscription not found' } });
+		} else {
+			response.send(res);
+		}
+	} catch (/** @type {any} */ e) {
+		/** @type {Object<string, string>} */
+		const errors = {};
+		logger.warn(e, 'unhandled error');
+		errors.unknown = e.message || 'unknown';
```

Comment:
Maybe this is a pattern we're using elsewhere, but in general we might consider exception contents to be sensitive - and so don't return them outside of the application

Reply:
Yeah I think in other places we do return them (remember the issue around the prisma error being printed down a web page?), but I agree we shouldn't. I will just put in a generic message instead.